lib/active_storage/blob_io.rb

```diff
@@ -0,0 +1,50 @@
+# Acts as a proxy to turn ActiveStorage file into IO object
+
+module FormatParser
+  module ActiveStorage
+    class BlobIO
+      # @param blob[ActiveStorage::Blob] the file with linked service
+      # @return [BlobIO]
+      def initialize(blob)
+        @blob = blob
+        @service = blob.service
+        @pos = 0
+      end
+
+      # Emulates IO#read, but requires the number of bytes to read.
+      # Rely on `ActiveStorage::Service.download_chunk` of each hosting type (local, S3, Azure, etc)
+      #
+      # @param n_bytes[Integer] how many bytes to read
+      # @return [String] the read bytes
+      def read(n_bytes)
+        http_range = (@pos..(@pos + n_bytes))
+        body = @service.download_chunk(@blob.key, http_range)
+        @pos += body.bytesize
+        body.force_encoding(Encoding::ASCII_8BIT)
```

Comment:
I think it is a good idea for us to do it always, because the encoding of the string we get depends on the SDK that is used by ActiveStorage or (in case of file access) of the `Encoding.default_external` - while all of the parsers expect binary.